File: crates/re_types/src/datatypes/class_description_ext.rs
use super::{AnnotationInfo, ClassDescription, Color};

impl From<(u16, &str)> for ClassDescription {
    fn from(value: (u16, &str)) -> Self {
        Self {
            info: value.into(),
            ..Default::default()
        }
    }
}

impl From<(u16, &str, Color)> for ClassDescription {
    fn from(value: (u16, &str, Color)) -> Self {
        Self {
            info: value.into(),
            ..Default::default()
        }
    }
}

impl From<AnnotationInfo> for ClassDescription {
    fn from(info: AnnotationInfo) -> Self {
        Self {
            info,
            ..Default::default()
        }
    }
}

File: rerun_cpp/src/rerun/datatypes/class_description_ext.cpp
#include "class_description.hpp"

namespace rerun::datatypes {

#if 0
    // <CODEGEN_COPY_TO_HEADER>

    /// Create a new `ClassDescription` from a single annotation info.
    ClassDescription(
        uint16_t id, std::optional<std::string> label = std::nullopt,
        std::optional<datatypes::Rgba32> color = std::nullopt
    )
        : info(id, label, color) {}

    ClassDescription(
        AnnotationInfo info_, Collection<AnnotationInfo> keypoint_annotations_ = {},
        Collection<KeypointPair> keypoint_connections_ = {}
    )
        : info(std::move(info_)),
            keypoint_annotations(std::move(keypoint_annotations_)),
            keypoint_connections(std::move(keypoint_connections_)) {}

    // </CODEGEN_COPY_TO_HEADER>
#endif

} // namespace rerun::datatypes
